### PR TITLE
Refactor: new class "BetaRadials" and "RadialCollection" for module_nao

### PR DESCRIPTION
**Pull request by jinzx10**
The new class "BetaRadials" is a concrete class derived from RadialSet, which is the counterpart of "Numerical_Nonlocal" in module_ao.

The new class "RadialCollection" is the top-level container for numerical radial functions. It is supposed to hold all radial functions of a certain kind (orbital or beta function), i.e., all numerical atomic orbitals or all beta functions.

### CODE REVIEW

File: source/module_basis/module_nao/radial_collection.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    void build(const int nfile, const std::string* const file, const char type = 'o');
+
+    /*! @name Getters
+     *
+     *  Get access to private members (and their properties).
+     *                                                                      */
+    //!@{
+    //! element symbol of a given type
+    const std::string& symbol(int itype) const { return radset_[itype]->symbol(); }
```
Comment (mohanchen) [inline source/module_basis/module_nao/radial_collection.h]: const int itype?

File: source/module_basis/module_nao/radial_set.cpp [hunk 2]
```diff
@@ -1,19 +1,147 @@
...
+        index_map_ = new int[(lmax_ + 1) * nzeta_max_];
+        for (int i = 0; i < (lmax_ + 1) * nzeta_max_; i++)
+        {
+            index_map_[i] = rhs.index_map_[i];
+        }
+    }
+    return *this;
+}
+
 int RadialSet::index(int l, int izeta) const
```
Comment (mohanchen) [inline source/module_basis/module_nao/radial_set.cpp]: const int l, const int izeta?

**Review by mohanchen — APPROVED** [review]

File: source/module_basis/module_nao/radial_collection.h (hunk 1 above)
Comment (jinzx10) [reply source/module_basis/module_nao/radial_collection.h]: const qualifiers have been appended

File: source/module_basis/module_nao/radial_set.cpp (hunk 2 above)
Comment (jinzx10) [reply source/module_basis/module_nao/radial_set.cpp]: const qualifiers have been appended

**Review by mohanchen — APPROVED** [review]

**Pull request merged**